modules/registry-accessing-directly.adoc

```diff
@@ -7,54 +7,53 @@
 
 You can access the registry directly to invoke `podman` commands. This allows
 you to push images to or pull them from the integrated registry directly using
-operations like `podman-push` or `podman-pull`. To do so, you must be logged in
-to the registry using the `podman-login` command. The operations you can perform
+operations like `podman push` or `podman pull`. To do so, you must be logged in
+to the registry using the `podman login` command. The operations you can perform
 depend on your user permissions, as described in the following sections.
 
 .Prerequisites
 
-* For any direct access, you must have a regular user for your preferred identity
-provider.
-** A regular user can generate an access token required for logging in to
-the registry.
-** System users, such as *system:admin*, cannot obtain access tokens
-and, therefore, cannot access the registry directly.
-+
-For example, if you are using `HTPASSWD` authentication, you can create one
-using the following command:
-+
-----
-# htpasswd /etc/origin/openshift-htpasswd <user_name>
-----
-
-* For pulling images, for example when using the `podman-pull` command,
-the user must have the *registry-viewer* role. To add this role:
+* You must have cluster administrator privileges.
+* You must have configured an identity provider (IDP).
+* For pulling images, for example when using the `podman pull` command,
+the user must have the `registry-viewer` role. To add this role:
 +
 ----
 $ oc policy add-role-to-user registry-viewer <user_name>
 ----
 
-* For writing or pushing images, for example when using the `podman-push` command,
-the user must have the *registry-editor* role. To add this role:
+* For writing or pushing images, for example when using the `podman push` command,
+the user must have the `registry-editor` role. To add this role:
 +
 ----
 $ oc policy add-role-to-user registry-editor <user_name>
 ----
 
 .Procedure
 
-. Log in to the registry directly:
-
-.. Ensure you are logged in to {product-title} as a *regular user*:
+. Ensure you are logged in to {product-title} as a cluster administrator:
 +
 ----
 $ oc login
 ----
 
-.. Log in to the container image registry by using your access token:
+. Look up the route for the registry from the `openshift-image-registry`
+namespace:
++
+----
+$ oc get routes -n openshift-image-registry
+----
++
+[NOTE]
+====
+Since the Image Registry Operator creates the route, it will likely be the
+`default-route`.
+====
+
+. Log in to the container image registry by using your access token:
 +
 ----
-$ podman-login -u openshift -p $(oc whoami -t) <registry_ip>:<port>
+$ podman login -u openshift -p $(oc whoami -t) $(oc get route -n openshift-image-registry default-route -o jsonpath='{.status.ingress[*].host}{"\n"}')
```

Comment:
If use podman to login, I suggest to use internal registry address like before: 
sh-4.4# podman login -u kubeadmin -p RDhLYKRgyw3JFpG4-9N9yBbLdGK4CtKeAc5aSjFuk3E image-registry.openshift-image-registry.svc:5000
Login Succeeded!

Reply:
If user external default address, user need to create a CA file to copy to local like this:
a.Get wildcard route ca
$oc get secret router-certs-default -n openshift-ingress -o yaml
$echo <tls.crt_content> | base64 -d > ca.crt
b. Trust ca in your client platform
$sudo cp ca.crt  /etc/pki/ca-trust/source/anchors/<external_route>.crt
#sudo update-ca-trust enable
c. Reload docker
#sudo systemctl daemon-reload
#sudo systemctl restart docker
d.the user can use docker to login with this default address: default-route-openshift-image-registry.<cluster_name>

Reply:
@dmage could you please confirm my above comment?

Reply:
Above steps can be added to accessing-the-registry.adoc.

Reply:
The assumptions of this article are not clear. If it assumes that all actions will be from outside of the cluster, then only external routes can be used. That's a good point that most likely the registry will have untrusted certificate.

Reply:
@bmcelvee so the conclusion is user has two options to access Openshift internal registry as below:
1. If use internal route, user has to be inside cluster like below:
a. Access into node:
$oc get nodes   --------------this is to get node address
$oc debug nodes/<node_address>
b. Use podman command to log into internal route: image-registry.openshift-image-registry.svc:5000:
sh-4.4# podman login -u kubeadmin -p RDhLYKRgyw3JFpG4-9N9yBbLdGK4CtKeAc5aSjFuk3E image-registry.openshift-image-registry.svc:5000
Login Succeeded!
c. Then you can use podman pull/push;

2. If use external routes, user need to enable defaultRoute=true, then create a CA file to copy to local like this:
a.Get wildcard route ca
$oc get secret router-certs-default -n openshift-ingress -o yaml
$echo <tls.crt_content> | base64 -d > ca.crt
b. Trust ca in your client platform
$sudo cp ca.crt /etc/pki/ca-trust/source/anchors/<external_route>.crt
#sudo update-ca-trust enable
c. Reload docker
#sudo systemctl daemon-reload
#sudo systemctl restart docker
d.the user can use docker to login with this default address locally: default-route-openshift-image-registry.<cluster_name>

Reply:
Thanks so much, @wzheng1! Wildcard routes are not supported in 4.1, so I've documented the external route method locally and can put it in the documentation if wildcard routes are offered in the future .

Reply:
@bmcelvee By re-reviewing the doc registry-accessing-directly.adoc, my above 2 methods are separated with each other, one is with internal route address: image-registry.openshift-image-registry.svc:5000 ; another is to login with external route address enabled by user: 
$ oc get route -n openshift-image-registry default-route -o jsonpath='{.status.ingress[*].host}{"\n"}'
default-route-openshift-image-registry.apps.qe-wewang-527.qe.devcluster.openshift.com;
If user use external route address, it is OK to use docker command.

Reply:
Thanks @wzheng1! So far I've only included the steps for accessing internally since wildcard routing is not supported in 4.1.

I'm a bit confused on the command we should document for the log in step, would it be better as one of the following:
1. $ podman login -u openshift -p $(oc whoami -t) $(oc get route -n openshift-image-registry default-route -o jsonpath='{.status.ingress[*].host}{"\n"}')

2. podman login -u kubeadmin -p RDhLYKRgyw3JFpG4-9N9yBbLdGK4CtKeAc5aSjFuk3E image-registry.openshift-image-registry.svc:5000

3. podman login -u kubeadmin -p $(oc whoami -t) image-registry.openshift-image-registry.svc:5000

Reply:
I vote for item #3.